backend/app/Usecases/Tenant/CreateItemAction.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Usecases\Tenant;
+
+use App\Constants\MessageConst;
+use App\Models\Item;
+use App\Models\Tenant;
+use App\Usecases\Tenant\Exceptions\UploadFileFailedException;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Str;
+
+class CreateItemAction
+{
+    public function __invoke(Tenant $tenant, Item $item, UploadedFile $image): Item
+    {
+        $item->tenant_id = $tenant->id;
+
+        try{
+            $path = 'tenants/' . $tenant->id . '/items';
+            $uniqueId = Str::uuid();
+            $extension = $image->extension();
+            $fileName = $uniqueId . '.' . $extension;
+            $filePath = $path . '/' . $fileName;
+
+            $result = Storage::disk('s3')->put($filePath, file_get_contents($image), 'public');
+            if(!$result) {
```

Comment:
割と細かいところですいません。
phpの比較はちょっと注意が必要で
https://www.php.net/manual/ja/types.comparisons.php
>bool : if($x)

とあるように

- $result = "-1"でif($result) →　true
- $result = nullでif($result) →　false

のようにわりとなんでも判定できてしまって、意図しない挙動になってしまうことがあります。

なので、｢=== による厳密な比較｣を行うのが良いかなと思います

Reply:
https://laravel.com/api/11.x/Illuminate/Support/Facades/Storage.html#method_put

らしいので失敗したらfalseが返ってきそう
つまり↓

```suggestion
            if($result === false) {
```